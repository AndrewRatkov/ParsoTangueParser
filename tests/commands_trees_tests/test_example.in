int x := 1+2+3+4+("a"!="b"*4)!=7;
int y := x + x;
str z := y * "aba" + "3" * (x + 2);
if x == y then
z := z + "BAD";
else
z:= z + "GOOD";
fi
y := 0;