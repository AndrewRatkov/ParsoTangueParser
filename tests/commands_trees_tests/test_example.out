[DEF int x]--->[INT,!=]------->[INT,+ ]------->[INT,+ ]------->[INT,+ ]------->[INT,+ ]------->[INT,1]
                   |               |               |               |               |
                   |               |               |               |               +---------->[INT,2]
                   |               |               |               |
                   |               |               |               +---------->[INT,3]
                   |               |               |
                   |               |               +---------->[INT,4]
                   |               |
                   |               +---------->[INT,!=]------->[STR,"a"]
                   |                               |
                   |                               +---------->[STR,* ]------->[STR,"b"]
                   |                                               |
                   |                                               +---------->[INT,4]
                   |
                   +---------->[INT,7]
[DEF int y]--->[INT,+ ]------->[INT,x]
                   |
                   +---------->[INT,x]
[DEF str z]--->[STR,+ ]------->[STR,* ]------->[INT,y]
                   |               |
                   |               +---------->[STR,"aba"]
                   |
                   +---------->[STR,* ]------->[STR,"3"]
                                   |
                                   +---------->[INT,+ ]------->[INT,x]
                                                   |
                                                   +---------->[INT,2]
IF
|
+--->[INT,==]------->[INT,x]
|        |
|        +---------->[INT,y]
|
THEN
|
+--->[DEF str z]--->[STR,+ ]------->[STR,z]
|                       |
|                       +---------->[STR,"BAD"]
|
ELSE
|
+--->[DEF str z]--->[STR,+ ]------->[STR,z]
|                       |
|                       +---------->[STR,"GOOD"]
|
FI
[DEF int y]--->[INT,0]
